dotCMS/src/main/java/com/dotcms/contenttype/business/ContentTypeDestroyAPIImpl.java

```diff
@@ -68,32 +130,53 @@ public void relocateContentletsForDeletion(final ContentType source, final Conte
             );
         }
 
-        final DotConnect dotConnect = new DotConnect();
+        final String tempTableName = String.format("contentlets_to_be_updated_%d", System.currentTimeMillis());
 
-        final String selectContentlets = "select c.inode from contentlet c where c.structure_inode =  ? \n";
-        dotConnect.setSQL(selectContentlets).addParam(source.inode());
+        //The quickest way to relocate contentlets is to create a temp table with the inodes of the contentlets to be relocated
+        final String dumpContentletToUpdate = dumpTableScript(tempTableName, source);
 
-        String updateContentlet = String.format(
-                "update contentlet c set structure_inode = ?, \n" +
-                        " contentlet_as_json = jsonb_set(contentlet_as_json,'{contentType}', '\"%s\"'::jsonb, false)  \n" +
-                        " where c.structure_inode =  ? ", target.inode()
-        );
+        Try.of(()->new DotConnect().executeStatement(dumpContentletToUpdate)).onFailure(e->{
```

Comment:
I do not understand why not using WrapInTransaction or handling the connections

Reply:
yeah, I removed it because in this particular situation, the method is getting called from another transactional context therefore we had said that here it wasn't really needed. but I'm adding it back for clarity